apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/AbstractRandomClaimGeneratorTest.java

```diff
@@ -0,0 +1,227 @@
+package gov.cms.bfd.pipeline.rda.grpc.server;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.Test;
+
+class AbstractRandomClaimGeneratorTest {
+
+  /** The fully developed set of random data that could potentially be produced. */
+  private static final Map<String, Object> expected =
+      Map.of(
+          "randomInt", 32,
+          "randomNumericString", "544365311023",
+          "randomAlphaNumericString", "fh3kxnp73k4s",
+          "randomEnum", TestGenerator.MyEnums.ONE,
+          "randomCharacter", 'd',
+          "randomAlphaString", "ddpwfcrncqsb",
+          "randomDate", "2021-05-05",
+          "randomAmount", "411.57");
+
+  /**
+   * Check that for any given configuration for a generated claim, the generated claims follow a
+   * stable random value generation pattern.
+   *
+   * <p>The test checks that altering the min/max values for a string just changes the length, and
+   * not the values themselves (aside from obvious truncation)
+   *
+   * <p>This test also checks that setting a value using always(), optional(), or even not setting
+   * the field at all also has no affect on the other generated field values
+   */
+  @Test
+  void changingAttributesTest() {
+    Clock clock = Clock.fixed(Instant.ofEpochMilli(1625172944844L), ZoneOffset.UTC);
+
+    // Try all the different field combinations
+    for (int i = Byte.MIN_VALUE; i <= Byte.MAX_VALUE; ++i) {
+      // Try always(), optional(), and even skipping the field entirely
+      for (int j = 0; j < TestGenerator.TestingState.values().length; ++j) {
```

Comment:
Rename `i` to `fieldMask` and instead of using `j` just loop over `TestingState.values() using a foreach loop.

```
      for (TestGenerator.TestingState value : TestGenerator.TestingState.values()) {
```